components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -142,30 +147,56 @@ - (UIColor *)titleColorForState:(UIControlState)state {
 
 #pragma mark - UIView
 
-// TODO(#7645): Temporary layout until
-// https://github.com/material-components/material-components-ios/issues/7645 lands
 - (void)layoutSubviews {
   [super layoutSubviews];
 
-  if (self.items.count == 0) {
-    return;
+  CGFloat availableWidth = CGRectGetWidth(self.bounds);
+  CGFloat maxWidth = 0;
+  for (UIView *itemView in self.containerView.arrangedSubviews) {
+    CGSize contentSize = itemView.intrinsicContentSize;
+    if (contentSize.width > maxWidth) {
+      maxWidth = contentSize.width;
+    }
   }
-  CGSize boundsSize = CGSizeMake(CGRectGetWidth(self.bounds), CGRectGetHeight(self.bounds));
-  CGSize itemSize = CGSizeMake(boundsSize.width / self.items.count, boundsSize.height);
+  CGFloat requiredWidth = maxWidth * self.items.count;
+  BOOL canBeJustified = availableWidth >= requiredWidth;
+  self.containerView.distribution = canBeJustified ? UIStackViewDistributionFillEqually
+                                                   : UIStackViewDistributionFillProportionally;
+}
 
-  CGPoint itemOrigin = CGPointMake(CGRectGetMinX(self.bounds), CGRectGetMinY(self.bounds));
-  for (UIView *itemView in self.itemViews) {
-    itemView.frame = CGRectMake(itemOrigin.x, itemOrigin.y, itemSize.width, itemSize.height);
-    itemOrigin = CGPointMake(itemOrigin.x + itemSize.width, itemOrigin.y);
+- (void)updateConstraints {
+  if (self.containerViewConstraintsActive) {
+    [super updateConstraints];
+    return;
   }
+
+  [_containerView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor].active = YES;
+  [_containerView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor].active = YES;
+  [_containerView.widthAnchor constraintGreaterThanOrEqualToAnchor:self.widthAnchor].active = YES;
+  [_containerView.topAnchor constraintEqualToAnchor:self.topAnchor].active = YES;
+  [_containerView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor].active = YES;
+  self.containerViewConstraintsActive = YES;
+
+  // Must always be called last according to the documentation.
+  [super updateConstraints];
 }
 
 - (CGSize)intrinsicContentSize {
-  return CGSizeMake(UIViewNoIntrinsicMetric, kMinHeight);
+  CGFloat totalWidth = 0;
+  CGFloat maxHeight = 0;
+  for (UIView *itemView in self.containerView.arrangedSubviews) {
+    CGSize contentSize = itemView.intrinsicContentSize;
+    totalWidth += contentSize.width;
+    if (contentSize.height > maxHeight) {
+      maxHeight = contentSize.height;
+    }
+  }
+  return CGSizeMake(totalWidth, MAX(kMinHeight, maxHeight));
 }
 
 - (CGSize)sizeThatFits:(CGSize)size {
```

Comment:
I also mentioned it in another thread, but please add unit tests for `sizeThatFits:`

Reply:
Added.